05_stringer/joshcarp/IPAddr_test.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var stdout = os.Stdout
+
+var tests = []struct {
+	input    IPAddr
+	expected string
+}{
+	{IPAddr{127, 0, 0, 1}, "127.0.0.1"},
+	{IPAddr{255, 255, 255, 255}, "255.255.255.255"},
+	{IPAddr{0, 0, 0, 0}, "0.0.0.0"},
+}
+
+func TestMain(t *testing.T) {
+	expected := "127.0.0.1\n"
+	r, w := captureStart()
+	main()
+	result := captureStop(r, w)
+	require.Equal(t, expected, result)
+}
+
+func TestStringer(t *testing.T) {
+	for _, test := range tests {
+		require.Equal(t, test.expected, test.input.String())
+	}
+}
+
+// captureStart diverts stdio to another file object
+func captureStart() (io.Reader, io.Closer) {
```

Comment:
A lot of people are using `fmt.Fprintln()` in their main function to make this process much simpler. This is cool though.